### PR TITLE
AT-1589: add run_checks method to PipelineExecutor

### PR DESCRIPTION
**Pull request by bluefloyd00**
Refactored:
- PipelineExecutor tests
- yaml structure
- added **kwargs to pygyver/etl/dw.py assert_unique()

added:
- run_checks(batch) method
- add run_checks to run_batch()

### CODE REVIEW

File: pygyver/etl/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Class BigQueryExecutor
+
+BigQuery functions wrap `bigpyquery` functions to provide higher level API removing boilerplate instructions of the lower level API.
```
Comment (charlielewis) [inline pygyver/etl/README.md:3]: bigpyquery?
google.cloud.bigquery is the the main client

File: pygyver/etl/dw.py [hunk 2]
```diff
@@ -734,7 +734,7 @@ def count_duplicates(self, table_id, primary_key: list, dataset_id=bq_default_da
         )
         return data['dup_total'].values[0]
 
-    def assert_unique(self, table_id, primary_key: list, dataset_id=bq_default_dataset(), ignore_error=False):
+    def assert_unique(self, table_id, primary_key: set, dataset_id=bq_default_dataset(), ignore_error=False, **kwargs):
```
Comment (charlielewis) [inline pygyver/etl/dw.py:737]: This is a reversion -- list is preferable to set

Comment (bluefloyd00) [reply pygyver/etl/dw.py:737]: my bad

File: pygyver/etl/pipeline.py [hunk 3]
```diff
@@ -31,18 +45,30 @@ def __init__(self, yaml_file):
         self.bq = BigQueryExecutor()
 
     def create_tables(self, batch):
-        batch_content = batch.get('batch','')
-        args = extract_args(batch_content, 'table')
+        batch_content = batch.get('tables', '')
```
Comment (charlielewis) [inline pygyver/etl/pipeline.py:48]: This works fine but I think it'd be a bit clearer to call it tables_content

File: pygyver/etl/pipeline.py [hunk 4]
```diff
@@ -31,18 +45,30 @@ def __init__(self, yaml_file):
...
+        result = execute_parallel(
+                    self.bq.create_table,
+                    args,
+                    message='Creating table:',
+                    log='table_id'
+                    )
+        return result
+
+    def run_checks(self, batch):
+        batch_content = batch.get('tables', '')
```
Comment (charlielewis) [inline pygyver/etl/pipeline.py:61]: I suggest tables_content is clearer

File: tests/test_pipeline.py [hunk 5]
```diff
@@ -46,17 +89,200 @@ def test_run_queries(self):
...
+                }
+            ]
+        }
+        try:
+            self.p_ex.run_checks(batch)
+        except AssertionError:
+            self.fail("run_checks() raised AssertionError unexpectedly!")
+
+    def tearDown(self):
+        if self.bq_client.table_exists(table_id='table1', dataset_id='test'):
```
Comment (charlielewis) [inline tests/test_pipeline.py:239]: Not sure if it's better to use BigqueryExecutor().delete_table() from Pygyver. Then you wouldn't have to test the condition.

But up to you.

**Pull request merged**